File: examples/python/tts_client.py
import requests

response = requests.post(
    "http://localhost:7777/v1/audio/speech",
    json={"input": "Hello from Python", "voice": "af_heart"},
    timeout=30,
)
response.raise_for_status()

with open("hello.wav", "wb") as handle:
    handle.write(response.content)

print("Wrote hello.wav")

File: examples/javascript/tts_client.js
async function playKokoro() {
  const response = await fetch("http://localhost:7777/v1/audio/speech", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ input: "Hello from the browser", voice: "af_heart" }),
  });

  if (!response.ok) {
    throw new Error(`kokoro-edge request failed: ${response.status}`);
  }

  const blob = await response.blob();
  const url = URL.createObjectURL(blob);
  const audio = new Audio(url);
  audio.play();
}

playKokoro().catch((error) => console.error(error));
